src/ngraph_builder.cc

```diff
@@ -3870,6 +3870,34 @@ static Status TranslateTransposeOp(
   TF_RETURN_IF_ERROR(
       GetStaticInputVector(op, 1, static_input_map, &permutation));
 
+  auto ng_input_rank = ng_input->get_shape().size();
+  std::unordered_set<int> axes;
+  for (auto p : permutation) {
+    if (p < ng_input_rank && !axes.count(p)) {
+      axes.insert(p);
+    } else {
+      auto missing_axes = 0;
+      for (int i = 0; i < ng_input_rank; i++) {
+        if (std::find(permutation.begin(), permutation.end(), i) ==
+            permutation.end()) {
+          missing_axes = i;
+          break;
+        }
+      }
+
+      std::ostringstream oss;
+      // Convert all but the last element to avoid a trailing ","
+      std::copy(permutation.begin(), permutation.end() - 1,
+                std::ostream_iterator<int>(oss, ", "));
+
+      // Now add the last element with no delimiter
+      oss << permutation.back();
+
+      return errors::InvalidArgument(missing_axes, " is missing from {",
+                                     oss.str(), "}.");
```

Comment:
Can we transpose an empty tensor, like a const?
The question being, does TF catch that in construction time or in runtime. If the error is runtime, we should be throwing the error.

Reply:
@sayantan-nervana Regarding this, I tried few cases, sharing my observations. 
```
a. transpose a const (permutation is an empty vector)
b. permutation vector contains numbers outside 0 to rank-1 (negative indexing?)
```

a. transpose a const (permutation is an empty vector)
   - with const input and perm vector [2] or [0] or [1] : we get a construction time error
   - with const input and empty perm vector [] : the test passes added a C++ unit test
b. permutation vector contains numbers outside 0 to rank-1 (negative indexing?)
     - if the size of the permutation vector is other than the rank of input : we get a construction error
- if the perm vector is the same size and either a negative value or number>input_rank is one of the perm vector : eg rank =3, perm = [-3,0,1] or [0,1,12] or [-111, 0,1]
 our check detects it as one of the dimension missing from the perm vector . So this is handled
- for some values (while testing negative indexes) i randomly see seg fault in tf itself (much before the execution reaches ngraph),  which we cant do anything about.